### PR TITLE
Radhaksha assignment2

### PR DESCRIPTION
**Pull request by radhaksha**
Hi @omontalv ,

I have submitted updated Assignment 2. Below are the updates:

1) Added GuitarSpec class.
2) Modified other classes as required.
3) Updated JUnit tests

Please review!

Thanks
Radha

### CODE REVIEW

File: src/FindGuitarTester.java [hunk 1]
```diff
@@ -1,47 +1,107 @@
...
+ * This class is a driver class that drives the application.
+ * It has main method which we use to search guitar of specific attributes.
+ * initializeInventory method initialize the guitar inventory with specific guitar attributes.
+ */
+
 public class FindGuitarTester {
 
+	/**
+	 * main method. entry point for the program.
+	 * @param args
```
Comment (omontalv) [inline src/FindGuitarTester.java]: @param args are ignored in this methode

File: src/FindGuitarTester.java [hunk 2]
```diff
@@ -1,47 +1,107 @@
...
+      }
+    	  
+  } 	
+    
+
+  
+  
+  /**
+   * 
+   * @param inventory. Loads all the provided guitar types in guitar's list.
```
Comment (omontalv) [inline src/FindGuitarTester.java]: Good comment, but the @param should be below the description.

File: src/Guitar.java [hunk 3]
```diff
@@ -1,43 +1,132 @@
...
+		  Gibson,
+		  Collings,
+		  Olson,
+		  Ryan,
+		  PRS
+	  }
+	  
+	  /**
+	   * This is a parameterized constructor.
+	   * @param serialNumber
```
Comment (omontalv) [inline src/Guitar.java]: comment on each param please.

File: src/GuitarSpec.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
+
```
Comment (omontalv) [inline src/GuitarSpec.java]: File header?

File: src/GuitarSpec.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+public class GuitarSpec {
```
Comment (omontalv) [inline src/GuitarSpec.java]: Class comment?

File: src/GuitarSpec.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+	  private String model;
+	  Guitar.backWood backwood_type;
+	  Guitar.topWood topwood_type;
+	  Guitar.type guitar_type;
+	  Guitar.builder guitar_builder;
+	
+	  
+	  /**
+	   * This is a parameterized constructor.
+	   * @param serialNumber
```
Comment (omontalv) [inline src/GuitarSpec.java]: Add a comment for each param.

File: src/GuitarSpec.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+	    this.backwood_type = backwood_type;
+	    this.topwood_type = topwood_type;
+	  }
+	
+	  
+	  
+
+
+
+	  public boolean matches(GuitarSpec otherspec) {
```
Comment (omontalv) [inline src/GuitarSpec.java]: Comment?

File: src/Inventory.java [hunk 8]
```diff
@@ -1,57 +1,137 @@
...
+
+  
+	public Inventory() {
+		guitars = new LinkedList<Guitar>();
+	}
+
+  
+	/**
+	 * This method adds new guitars to the list guitars.
+	 * @param serialNumber
```
Comment (omontalv) [inline src/Inventory.java]: Comment for each param

File: src/Inventory.java [hunk 9]
```diff
@@ -1,57 +1,137 @@
...
+	public void addGuitar(String serialNumber, double price, Guitar.builder guitar_builder, String model, Guitar.type guitar_type, Guitar.backWood backwood_type, Guitar.topWood topwood_type)  {		
+		Guitar guitar = new Guitar(serialNumber, price, guitar_builder, model, guitar_type, backwood_type, topwood_type);		
+		guitars.add(guitar);
+	}
+  	
+
+	
+
+	/**
+	 *
```
Comment (omontalv) [inline src/Inventory.java]: Please add description before params.

File: src/Inventory.java [hunk 10]
```diff
@@ -1,57 +1,137 @@
...
+				return guitar;
+			}		
+    }		 
+	return null;  
+	}
+  	
+	
+	/**
+	 * 
+	 * @param searchGuitar. Checks for builder, model, type, backWood and topWood of the provided guitar.
```
Comment (omontalv) [inline src/Inventory.java]: Description before params? This should also describe how the search/compare is done. Does it match things exactly? Are there wildcards?

File: src/Inventory.java [hunk 11]
```diff
@@ -1,57 +1,137 @@
...
+		guitars.add(guitar);
+	}
+  	
+
+	
+
+	/**
+	 * 
+	 * @param serialNumber. Takes serialNumber and returns guitar matching the provided serialNumber
+	 * @return guitar
```
Comment (omontalv) [inline src/Inventory.java]: What happens if guitar is not found? Does it return null? Throw an exception?

File: src/Inventory.java [hunk 12]
```diff
@@ -1,57 +1,137 @@
...
+			}		
+    }		 
+	return null;  
+	}
+  	
+	
+	/**
+	 * 
+	 * @param searchGuitar. Checks for builder, model, type, backWood and topWood of the provided guitar.
+	 * @return guitar
```
Comment (omontalv) [inline src/Inventory.java]: What happens if guitars is not found? Does it return null? Throw an exception?

Comment (radhaksha) [reply src/Inventory.java]: Hi @omontalv ,

If guitar is not found then program returns following output. It doesn't throw exception.

"Sorry, Erin, we have nothing for you."

**Comment by radhaksha** [comment]
@omontalv ,
 Hi Professor, I

**Pull request closed without merging**